metricbeat/helper/server/http/http.go

```diff
@@ -52,7 +58,15 @@ func (h *HttpEvent) GetMeta() server.Meta {
 	return h.meta
 }
 
-func NewHttpServer(mb mb.BaseMetricSet) (server.Server, error) {
+func (h *HttpEvent) SetEvent(event common.MapStr) {
+	h.event = event
+}
```

Comment:
Having that you are now passing a custom handler when initializing (which I think is a good idea), do you need an HttpEvent construct anymore? This was used to pass data between a generic HTTP server and the metricset. But now the metricset is in control of the requests, and can directly build and send events to the output. That should make things simpler, WDYT?